tests/test_iqm_naive_move_pass.py

```diff
@@ -1,85 +1,240 @@
+from qiskit.compiler import transpile
+from qiskit.quantum_info import Operator, Statevector
 
+from iqm.iqm_client import ExistingMoveHandlingOptions, QuantumArchitectureSpecification
+from iqm.qiskit_iqm.fake_backends import IQMErrorProfile, IQMFakeAdonis, IQMFakeAphrodite, IQMFakeBackend, IQMFakeDeneb
+from iqm.qiskit_iqm.iqm_circuit_validation import validate_circuit
 from iqm.qiskit_iqm.iqm_naive_move_pass import transpile_to_IQM
-from iqm.qiskit_iqm.iqm_provider import IQMBackend
+from iqm.qiskit_iqm.iqm_transpilation import IQMReplaceGateWithUnitaryPass
+from iqm.qiskit_iqm.move_gate import MOVE_GATE_UNITARY, MoveGate
 
-from .utils import _get_allowed_ops, _is_valid_instruction, get_mocked_backend
+from .utils import capture_submitted_circuits, get_mocked_backend
 
 
-def test_transpile_to_IQM_star_semantically_preserving(ndonis_architecture):  # pylint: disable=too-many-locals
-    backend, _client = get_mocked_backend(ndonis_architecture)
-    qubit_registers = _get_qubit_registers(backend)
-    n_qubits = len(qubit_registers)
-    for i in range(2, n_qubits + 1):
-        circuit = QuantumVolume(i, i)
-        # Use optimization_level=0 to avoid that the qubits get remapped.
-        transpiled_circuit = transpile_to_IQM(circuit, backend, optimization_level=0)
-        transpiled_operator = Operator(transpiled_circuit)
+def quantum_volume_circuits(sizes: Iterable[int] = range(2, 6)):
+    """Generate random quantum volume circuits for testing."""
+    return [QuantumVolume(n) for n in sizes]
 
-        # Update the original circuit to have the correct number of qubits and resonators.
-        original_with_resonator = QuantumCircuit(backend.num_qubits, 0)
-        original_with_resonator.append(circuit, [qubit_registers[circuit.find_bit(i)[0]] for i in circuit.qubits])
 
-        # Make it into an Operator and cheeck equivalence.
-        circuit_operator = Operator(original_with_resonator)
-        assert circuit_operator.equiv(transpiled_operator)
+def ghz_circuits(sizes: Iterable[int] = range(2, 6)):
+    """Generate GHZ circuits for testing."""
+    circuits = []
+    for n in sizes:
+        qc = QuantumCircuit(n)
+        qc.h(0)
+        # for i in range(n - 1):
+        #    qc.cx(i, i + 1)
+        for i in range(1, n):
+            qc.cx(0, i)
+        circuits.append(qc)
+    return circuits
+
+
+def hypothetical_fake_device():
+    """Generate a hypothetical fake device for testing.
+
+          QB1   QB2
+            |   |
+           COMP_R1
+            |   |
+    QB3 - QB4   QB7 - QB8
+            |   |
+           COMP_R2
+            |   |
+          QB5   QB6
+
+
+    """
+    architecture = QuantumArchitectureSpecification(
+        name="Hypothetical",
+        qubits=["COMP_R1", "COMP_R2", "QB1", "QB2", "QB3", "QB4", "QB5", "QB6", "QB7", "QB8"],
+        operations={
+            "prx": [["QB1"], ["QB2"], ["QB3"], ["QB4"], ["QB5"], ["QB6"], ["QB7"], ["QB8"]],
+            "cz": [
```

Comment:
There is no pair with only CZ and not MOVE gate, but this is the case we see on the real devices, some pairs might only have CZ calibrated without MOVE.

Reply:
There are many, the CZ is on much fewer pairs than the MOVEs. Also, QB3-QB4 and QB7-QB8 are instances where there is no COMP_R in between them.